ansible/roles/layers-service/tasks/main.yml

```diff
@@ -0,0 +1,93 @@
+- include: ../../common/tasks/setfacts.yml
+
+- name: install Postgis and GDAL Execs
+  apt: pkg={{ item }} state=present update_cache=yes
+  with_items:
+    - postgis
+    - gdal-bin
+
+- name: copy transient files to tmp (schemas etc)
+  copy: src={{item}} dest=/tmp
+  with_items:
+    - layersdb.sql
+    - stored_procs.sql    
+  tags:
+    - createdb
+
+# Not working currently.....
+- name: setup the DB - WARNING!!! this isnt idempotent yet
```

Comment:
There are dedicated postgresql tasks for creating databases and users. The problem with command tasks like this is that ansible always treats them as 'changed' so they run every time.

The postgresql_db and postgresql_user tasks are idempotent. They also allow you to register a variable to record whether they were executed or not: this can then be used to determine whether to execute the DB scripts below. E.g.:
```
- name: create database
  postgresql_db: name={{ volunteer_portal_db_name }}
  sudo_user: postgres
  sudo: True
  register: dbschema
  tags:
    - db
    - volunteer-portal
```
...